client/app/account/billing/common/renew-helper.service.js

```diff
@@ -1,65 +1,74 @@
-angular.module("Billing.services").service("BillingrenewHelper", function ($filter, $translate) {
-    "use strict";
-
-    this.getRenewDateFormated = (service) => {
-        if (!service) {
-            return "";
-        } else if (this.serviceHasAutomaticRenew(service)) {
-            return _.capitalize($filter("date")(service.expiration, "MMMM yyyy"));
+angular
+    .module("Billing.services")
+    .service("billingRenewHelper", class BillingrenewHelper {
+        constructor ($filter, $translate) {
+            this.$filter = $filter;
+            this.$translate = $translate;
         }
 
-        let translationId = "autorenew_service_expiration_date";
-        if (moment().isAfter(moment(service.expiration))) {
-            translationId = "autorenew_service_after_expiration_date";
-        }
-        return $translate.instant(translationId, { t0: $filter("date")(service.expiration, "mediumDate") });
-    };
-
-    this.getRenewLabel = (service) => {
-        if (service === "0") {
-            return $translate.instant("autorenew_service_renew_0");
-        }
+        getRenewDateFormated (service) {
+            if (!service) {
+                return "";
+            } else if (this.serviceHasAutomaticRenew(service)) {
+                return _.capitalize(this.$filter("date")(service.expiration, "MMMM yyyy"));
+            }
 
-        if (angular.isString(service)) {
-            if (service.indexOf("frequency_value_") > -1) {
-                return $filter("renewFrequence")(+service.split("_")[2]);
+            let translationId = "autorenew_service_expiration_date";
+            if (moment().isAfter(moment(service.expiration))) {
+                translationId = "autorenew_service_after_expiration_date";
             }
-            return $translate.instant(`autorenew_service_renew_${service}`);
+            return this.$translate.instant(translationId, { t0: this.$filter("date")(service.expiration, "mediumDate") });
         }
 
-        if (angular.isObject(service)) {
-            if (!_.isEmpty(service.subProducts)) {
-                return "";
+        getRenewLabel (service) {
+            if (service === "0") {
```

Comment:
Strange to test a number in string format :(

Reply:
Didn't refactor this